# Segmenting and Clustering Neighborhoods in Toronto

## 1. Scrape the Postal Codes of Canada from Wikipedia

#### Import the required libraries

In [1]:
import requests
import pandas as pd

#display all columns
pd.set_option('display.max_columns', None)

#### Define the url and send the get request

In [2]:
url = 'https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M'
canada_postal_codes = requests.get(url)

print(canada_postal_codes)

<Response [200]>


_The request was successful._

#### Read the postal codes data of canada from the html file received

In [3]:
column_names = ['Postal Code', 'Borough', 'Neighborhood']

canada_data = pd.read_html(canada_postal_codes.text, header = 0)
canada_data = canada_data[0]

canada_data.head()

Postal Code           Borough               Neighborhood
0         M1A      Not assigned                        NaN
1         M2A      Not assigned                        NaN
2         M3A        North York                  Parkwoods
3         M4A        North York           Victoria Village
4         M5A  Downtown Toronto  Regent Park, Harbourfront

#### Delete the rows with "Not assigned" Borough

In [4]:
index = canada_data[canada_data['Borough'] == 'Not assigned'].index
canada_data.drop(index, inplace=True)
canada_data.reset_index(drop = True, inplace = True)

canada_data.head()

Postal Code           Borough                                 Neighborhood
0         M3A        North York                                    Parkwoods
1         M4A        North York                             Victoria Village
2         M5A  Downtown Toronto                    Regent Park, Harbourfront
3         M6A        North York             Lawrence Manor, Lawrence Heights
4         M7A  Downtown Toronto  Queen's Park, Ontario Provincial Government

#### Postal code dataframe of Canada:

In [5]:
canada_data

Postal Code           Borough  \
0           M3A        North York   
1           M4A        North York   
2           M5A  Downtown Toronto   
3           M6A        North York   
4           M7A  Downtown Toronto   
5           M9A         Etobicoke   
6           M1B       Scarborough   
7           M3B        North York   
8           M4B         East York   
9           M5B  Downtown Toronto   
10          M6B        North York   
11          M9B         Etobicoke   
12          M1C       Scarborough   
13          M3C        North York   
14          M4C         East York   
15          M5C  Downtown Toronto   
16          M6C              York   
17          M9C         Etobicoke   
18          M1E       Scarborough   
19          M4E      East Toronto   
20          M5E  Downtown Toronto   
21          M6E              York   
22          M1G       Scarborough   
23          M4G         East York   
24          M5G  Downtown Toronto   
25          M6G  Downtown Toronto   
26          M1H       Scarborough   
27          M2H        North York   
28          M3H        North York   
29          M4H         East York   
..          ...               ...   
73          M4R   Central Toronto   
74          M5R   Central Toronto   
75          M6R      West Toronto   
76          M7R       Mississauga   
77          M9R         Etobicoke   
78          M1S       Scarborough   
79          M4S   Central Toronto   
80          M5S  Downtown Toronto   
81          M6S      West Toronto   
82          M1T       Scarborough   
83          M4T   Central Toronto   
84          M5T  Downtown Toronto   
85          M1V       Scarborough   
86          M4V   Central Toronto   
87          M5V  Downtown Toronto   
88          M8V         Etobicoke   
89          M9V         Etobicoke   
90          M1W       Scarborough   
91          M4W  Downtown Toronto   
92          M5W  Downtown Toronto   
93          M8W         Etobicoke   
94          M9W         Etobicoke   
95          M1X       Scarborough   
96          M4X  Downtown Toronto   
97          M5X  Downtown Toronto   
98          M8X         Etobicoke   
99          M4Y  Downtown Toronto   
100         M7Y      East Toronto   
101         M8Y         Etobicoke   
102         M8Z         Etobicoke   

                                          Neighborhood  
0                                            Parkwoods  
1                                     Victoria Village  
2                            Regent Park, Harbourfront  
3                     Lawrence Manor, Lawrence Heights  
4          Queen's Park, Ontario Provincial Government  
5              Islington Avenue, Humber Valley Village  
6                                       Malvern, Rouge  
7                                            Don Mills  
8                      Parkview Hill, Woodbine Gardens  
9                             Garden District, Ryerson  
10                                           Glencairn  
11   West Deane Park, Princess Gardens, Martin Grov...  
12              Rouge Hill, Port Union, Highland Creek  
13                                           Don Mills  
14                                    Woodbine Heights  
15                                      St. James Town  
16                                  Humewood-Cedarvale  
17   Eringate, Bloordale Gardens, Old Burnhamthorpe...  
18                   Guildwood, Morningside, West Hill  
19                                         The Beaches  
20                                         Berczy Park  
21                                 Caledonia-Fairbanks  
22                                              Woburn  
23                                             Leaside  
24                                  Central Bay Street  
25                                            Christie  
26                                           Cedarbrae  
27                                   Hillcrest Village  
28     Bathurst Manor, Wilson Heights, Downsview North  

In [6]:
print(canada_data.shape)

(103, 3)


__*There are a total of 103 rows (postal code data) in the dataframe*__